tool/tsh/tsh.go

```diff
@@ -939,6 +955,61 @@ func onBenchmark(cf *CLIConf) {
 		fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
 		os.Exit(255)
 	}
+	if cf.BenchExport {
+		var fullPath string
+		timeStamp := fmt.Sprint(time.Now().Format("2006-01-02_15:04:05"))
+		suffix := fmt.Sprint("/latency_profile_" + timeStamp + ".txt")
+		// If path is specified
+		if cf.BenchExportPath != "" {
+			usr, err := user.Current()
+			if err != nil {
+				fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+			}
+			dir := usr.HomeDir
+			// Expand tilde if used
+			if cf.BenchExportPath == "~" {
+				cf.BenchExportPath = dir
+			} else if strings.HasPrefix(cf.BenchExportPath, "~/") {
+				cf.BenchExportPath = filepath.Join(dir, cf.BenchExportPath[2:])
+			}
+			// Check if path is valid
+			if _, err := os.Stat(cf.BenchExportPath); err != nil {
+				if os.IsNotExist(err) {
+					fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+				}
+			}
+			// Remove trailing slash
+			if cf.BenchExportPath[len(cf.BenchExportPath)-1] == '/' {
+				cf.BenchExportPath = cf.BenchExportPath[0 : len(cf.BenchExportPath)-1]
+			}
+			fullPath = fmt.Sprint(cf.BenchExportPath + suffix)
+		} else {
+			fullPath = fmt.Sprint(client.FullProfilePath("") + suffix)
+		}
+
+		fo, err := os.Create(fullPath)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+			os.Exit(255)
+		}
+
+		w := bufio.NewWriter(fo)
+		defer func() {
+			if err := fo.Close(); err != nil {
+				fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+				return
+			}
+			fmt.Printf("Latency profile saved: %v", fullPath)
+			fmt.Printf("\n")
+		}()
+
+		_, err = result.Histogram.PercentilesPrint(w, cf.BenchTicks, cf.BenchValueScale)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, utils.UserMessageFromError(err))
+		}
+		w.Flush()
```

Comment:
Extract this into a separate function

Reply:
I agree with @awly , if you extract this logic in the separate function:

func writeHistogram(w io.Writer, h Histogram) error

you can later use and reuse it:

```go
f, err := ioutil.CreateFile()
...
if err := writeHistogram(f, histogram); err != nil {
    fmt.Fprintln(os.Stderr, trace.UserMessage(err))
}
```